doc/user/content/sql/alter-sink.md

```diff
@@ -0,0 +1,119 @@
+---
+title: "ALTER SINK"
+description: "`ALTER SINK` allows cutting a sink over to a new upstream relation without causing disruption to downstream consumers."
+menu:
+  main:
+    parent: 'commands'
+---
+
+`ALTER SINK` allows cutting a sink over to a new upstream relation without
+causing disruption to downstream consumers. This is useful in the context
+of [blue/green deployments](/manage/dbt/development-workflows/#bluegreen-deployments).
+
+## Syntax
+
+{{< diagram "alter-sink.svg" >}}
+
+## Details
+
+To alter the upstream relation a sink depends on and ensure continuity in data
+processing, Materialize must pick a consistent cutover timestamp. When you
+execute an `ALTER SINK` command, the resulting output will contain all the
+updates that happened before the cutover timestamp for the old relation, as
+well as all the updates that happened after the cutover timestamp for the new
+relation.
+
+{{< note >}}
+To select a consistent timestamp, Materialize waits for the previous definition
+of the sink to make enough progress such that there is overlap between the
+available timestamps of the newly specified upstream relation. Attempting to
+`ALTER` an unhealthy sink that can't make progress will result in the command
+timing out.
+{{</ note >}}
+
+### Valid schema changes
+
+For `ALTER SINK` to be successful, the newly specified relation must lead to a
+valid sink definition with the same conditions as the original `CREATE SINK`
+statement.
+
+When using the Avro format with a schema registry, the generated Avro
+schema for the new relation must be compatible with the previously published
+schema. If that's not the case, the `ALTER SINK` command will succeed, but the
+subsequent execution of the sink will result in errors and will not be able to
+make progress.
```

Comment:
Maybe add a recommendation here: monitor your sink status via <this SQL query> or in the console after an alteration to be ensure it remains healthy.